components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml

```diff
@@ -5590,6 +5590,18 @@ document.observe("xwiki:dom:updated", init);
     }
   });
 
+  var init = function(event) {
+    ((event &amp;&amp; event.memo.elements) || [$('body')]).each(function(element) {
+      element.select('input.check-unique, .external_id input').each(function(input) {
+        if (!input.__uniqueness_validator) {
+          var entity = $("entity-type") ? $("entity-type").value : "patients";
```

Comment:
This is relying on the hidden input with `id="entity-type"` defined by the access rights management UIX, correct? If that's the case, then external_id uniqueness won't be properly checked for families when the permissions manager is disabled (and we do have PhenoTips instalations configured this way at the moment). Now that we want to use that input in a new context, maybe it should be defined in the patient and family sheets instead of the permissions UIX?

Also, if we can't figure outthe entitytype I would just do nothing instead of defaulting to `patients`.

Reply:
The new entity resolver component can resolve entity type by document name (e.g.  P0000xxx or FAM000xxx). Maybe we should use that in one way or another?

And +1 to having entity-type always defined, regardless of optional component presence